dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -2446,17 +2436,15 @@ SimpleColumn* buildSimpleColFromDerivedTable(gp_walk_info& gwi, Item_field* ifp)
                     sc->alias(ifp->is_autogenerated_name() ? cols[j]->alias() : ifp->name.str);
                     sc->tableName(csep->derivedTbAlias());
                     sc->colPosition(j);
-                    string tableAlias(csep->derivedTbAlias());
-                    sc->tableAlias(lower(tableAlias));
+                    sc->tableAlias(csep->derivedTbAlias());
                     sc->timeZone(gwi.thd->variables.time_zone->get_name()->ptr());
                     if (!viewName.empty())
                     {
-                        // TODO: Remove lower when MCOL-4144 fixed    
-                        sc->viewName(lower(viewName));
+                        sc->viewName(viewName, lower_case_table_names);
                     }
                     else
                     {
-                            sc->viewName(lower(csep->derivedTbView()));
+                        sc->viewName(csep->derivedTbView());
```

Comment:
This should also have the `lower_case_table_names` as the second argument.

Reply:
All values put into csep have already been converted